### ISSUE TITLE
replace react-dnd with something else

### ISSUE THREAD

**Issue by macrozone** [react-page/react-page#1326]
we still use version 15 of react-dnd as version 16 does not work in our setup.

but version 15 also leads to build problems i different environments, see 

https://github.com/react-page/react-page/issues/1190
https://github.com/react-page/react-page/issues/1267
https://github.com/react-page/react-page/issues/1310

react-dnd did not receive any update since version 16 and the issues are starting to pile up, so maybe it got abandoned.

its therefore worth to look for alternatives

**dnd-kit**

https://github.com/clauderic/dnd-kit

has 10k stars and looks activily maintained. It does not use the html drag&drop api, but that should not be a big issue. React-page historically listend to certain drags from "outside" of the page, for example you could have dragged images from your desktop to react-page. But to be honest, i am not sure if that feature still works, i think i ditched it at some point because of maintainability. So this limitation should not be an issue.

are there other alternatives?

**Comment by bnowak008** [comment]
@macrozone Thanks for you're work on picking up and maintaining this package. I've recently begun to utilize this package but am looking into making some changes to allow me to utilize it along with vite and this is one of the bigger issues with it currently.

I currently have stripped out all references to the react-dnd package and have it compiling properly with vite. Before I put too much effort toward replacing the DND functionality, I'm curious if there is any work being done toward this currently? I know there is a beta package, but that does not seem to be functioning properly or actively committed too.

I'd like to either start contributing to this package to help with some of the issues or create a fork of my own that I can maintain without worrying about breaking changes. I would prefer to help with contributing to this package but depending on your level of interest in maintaining this project it may be better for myself to fork off of it

**Comment by macrozone** [comment]
> @macrozone Thanks for you're work on picking up and maintaining this package. I've recently begun to utilize this package but am looking into making some changes to allow me to utilize it along with vite and this is one of the bigger issues with it currently.
> 
> I currently have stripped out all references to the react-dnd package and have it compiling properly with vite. Before I put too much effort toward replacing the DND functionality, I'm curious if there is any work being done toward this currently? I know there is a beta package, but that does not seem to be functioning properly or actively committed too.
> 

no, I can't currently work on it, so contributions are very welcomed. But its not that easy to replace react-dnd though.

Maybe it would also be an option to fork or look for a forked version of react-dnd where the problems are fixed. i assume its some bundling error on their side.

> I'd like to either start contributing to this package to help with some of the issues or create a fork of my own that I can maintain without worrying about breaking changes. I would prefer to help with contributing to this package but depending on your level of interest in maintaining this project it may be better for myself to fork off of it

usually forking is bad, unless you contribute back. So if you want to contribute, you can open a pull request if you find a solution for react-dnd! It would be very welcomed
